Assets/Scripts/Player/UnitGridCombat.cs

```diff
@@ -147,6 +163,66 @@ private enum State {
 
 		public override void AttackUnit(CoreUnit unitGridCombat, Actions.AttackType type, Action onAttackComplete) {
 			state_ = State.Attacking;
+			shouldPlayAttackAnimation_ = true;
+			Vector3 attackerPosition = this.GetPosition();
+			Vector3 enemyPosition = unitGridCombat.GetPosition();
+			direction_ = Direction.Null;
+			int diffX = 0;
+			int diffY = 0;
+			// 0 is neither, 1 is yes, 2 is false.
+			int isBelow = 0;
+			int isLeft = 0;
+			if(attackerPosition.x == enemyPosition.x) {
+				// Either above or below us.
+			} else if(attackerPosition.x > enemyPosition.x) {
+				// Left of us.
+				diffX = (int)attackerPosition.x - (int)enemyPosition.x;

```

Comment:
For all the int casts here, nothing wrong about them but it might be better to do `(int)(attackerPosition.x - enemyPosition.x)`. But again I dont know for sure.

Reply:
Or like the rounding i have done in the sortingorder on line [23](https://github.com/Online-Pluisje-Art/Operation-Blackwell/blob/development/Assets/Scripts/Core/Utils/PositionSortingOrder.cs#L23).

Reply:
Works too, I'll change it.